spring/spring-web-reactive/src/main/java/io/quarkus/ts/spring/web/boostrap/web/BookController.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.ts.spring.web.boostrap.web;
+
+import static javax.ws.rs.core.Response.Status.CREATED;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.NO_CONTENT;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.quarkus.hibernate.reactive.panache.Panache;
+import io.quarkus.hibernate.reactive.panache.common.runtime.ReactiveTransactional;
+import io.quarkus.ts.spring.web.boostrap.persistence.model.Book;
+import io.quarkus.ts.spring.web.boostrap.persistence.repo.BookRepository;
+import io.quarkus.ts.spring.web.boostrap.web.exception.BookIdMismatchException;
+import io.quarkus.ts.spring.web.boostrap.web.exception.BookNotFoundException;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@RestController
+@RequestMapping("/api/books")
+@Produces("application/json")
+@Consumes("application/json")
+public class BookController {
+
+    private static final int FAIL_AFTER_SECONDS = 10;
+
+    @Autowired
+    BookRepository bookRepository;
+
+    @GetMapping
+    public Uni<List<Book>> findAll() {
+        return bookRepository.listAll().onItem()
+                .ifNull().continueWith(new ArrayList<Book>())
```

Comment:
`new ArrayList<Book>()` could be replaced by `Collections.emptyList()` (less verbose more semantic)